language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java

```diff
@@ -0,0 +1,146 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
+                                                    CodeActionContext context) {
+        if (!(diagnostic.message().contains(CommandConstants.INCOMPATIBLE_TYPES))) {
+            return Collections.emptyList();
+        }
+        List<DiagnosticProperty<?>> props = diagnostic.properties();
+        if (props.size() != 2 || props.get(FOUND_SYMBOL_INDEX).kind() != DiagnosticPropertyKind.SYMBOLIC) {
+            return Collections.emptyList();
+        }
+
+        Symbol rhsTypeSymbol = ((DiagnosticProperty<Symbol>) props.get(FOUND_SYMBOL_INDEX)).value();
+
+        // Skip, non-local var declarations
+        Node matchedNode = context.positionDetails().matchedNode();
+        if (matchedNode.kind() != SyntaxKind.LOCAL_VAR_DECL &&
+                matchedNode.kind() != SyntaxKind.MODULE_VAR_DECL &&
+                matchedNode.kind() != SyntaxKind.ASSIGNMENT_STATEMENT) {
+            return Collections.emptyList();
+        }
+
+        Optional<ExpressionNode> typeNode = getTypeNode(matchedNode, context);
+        Optional<String> variableName = getVariableName(matchedNode);
+        if (typeNode.isEmpty() || variableName.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        // Derive possible types
+        List<CodeAction> actions = new ArrayList<>();
+        List<TextEdit> importEdits = new ArrayList<>();
+        List<String> types = CodeActionUtil.getPossibleTypes((TypeSymbol) rhsTypeSymbol, importEdits, context);
+        for (String type : types) {
+            List<TextEdit> edits = new ArrayList<>();
+            edits.add(new TextEdit(CommonUtil.toRange(typeNode.get().lineRange()), type));
+            String commandTitle = String.format(CommandConstants.CHANGE_VAR_TYPE_TITLE, variableName.get(), type);
+            actions.add(createQuickFixCodeAction(commandTitle, edits, context.fileUri()));
+        }
+        return actions;
+    }
+
+    private Optional<ExpressionNode> getTypeNode(Node matchedNode, CodeActionContext context) {
+        switch (matchedNode.kind()) {
+            case LOCAL_VAR_DECL:
+                return Optional.of(
+                        ((VariableDeclarationNode) matchedNode).typedBindingPattern().typeDescriptor());
+            case ASSIGNMENT_STATEMENT:
+                Optional<VariableSymbol> optVariableSymbol = getVariableSymbol(context, matchedNode);
+                if (optVariableSymbol.isEmpty()) {
+                    return Optional.empty();
+                }
+                SyntaxTree syntaxTree = context.workspace().syntaxTree(context.filePath()).orElseThrow();
+                NonTerminalNode node = CommonUtil.findNode(optVariableSymbol.get(), syntaxTree);
+                if (node.kind() == SyntaxKind.TYPED_BINDING_PATTERN) {
+                    return Optional.of(((TypedBindingPatternNode) node).typeDescriptor());
+                } else {
```

Comment:
can we remove the else clause and just return?

Reply:
Yes we can do, but does it really matter?